### DISCUSSION TITLE
p2g link cluster info

### DISCUSSION THREAD

**Discussion by LacquerHed** [GreenleafLab/ArchR#1824] [Questions / Documentation]
Hi I am trying to retrieve list with clusters where p2g are enriched, doing the following but it does not contain cluster information per line:

```
p2geneDF <- metadata(proj@peakSet)$Peak2GeneLinks
p2geneDF$geneName <- mcols(metadata(p2geneDF)$geneSet)$name[p2geneDF$idxRNA]
p2geneDF$peakName <- (metadata(p2geneDF)$peakSet %>% {paste0(seqnames(.), "_", start(.), "_", end(.))})[p2geneDF$idxATAC]
```

**Comment by rcorces** [comment]
Peak to gene links are not called with respect to clusters, they are called based on variance across all cells.

**Reply by LacquerHed** [reply]
Ok thank you. I guess a more precise question is whether its possible to combine the information on peak2gene links and cluster marker peaks? To get an overlap of those two categories.

**Reply by LacquerHed** [reply]
I tried to do the following to just call links on specific cells and got an error:

```
proj <- addPeak2GeneLinks(Klf4_PEC_ArchR_subset_C5_13_14_integrated1, cellsToUse = Klf4_PEC_ArchR_subset_C5_13_14_integrated1$cellNames[Klf4_PEC_ArchR_subset_C5_13_14_integrated1$Clusters2=="PEC_0"])
p2gSub <- getPeak2GeneLinks(proj, corCutOff = 0.45, resolution = 1, returnLoops = TRUE)

Error in .validInput(input = start, name = "start", valid = c("integer")) : 
  Input value for 'start' is not a integer, (start = numeric) please supply valid input!
In addition: Warning message:
In min(abs(c(input%%1, input%%1 - 1)), na.rm = TRUE) :
  no non-missing arguments to min; returning Inf
```
addPeak2GeneLinks line runs fine but then I get an error when I try to retrieve the loops.
Saw that this was fixed a while back so not sure why Im getting it again?
Thanks.

**Reply by rcorces** [reply]
Something must be wrong with your peak set

**Reply by LacquerHed** [reply]
Any ideas on where I might look for what’s wrong with peakset?

**Reply by rcorces** [reply]
your error is coming from here https://github.com/GreenleafLab/ArchR/blob/55ab73efe716c0288fe5bb3edb063573d38ac8df/R/IntegrativeAnalysis.R#L1463-L1467
You can look at that code and try to understand why the value passed to `start` is not valid.
